db/db.go

```diff
@@ -11,66 +11,64 @@
 package db
 
 import (
-	"os"
+	"errors"
 
 	"github.com/Mierdin/todd/agent/defs"
 	"github.com/Mierdin/todd/config"
 	"github.com/Mierdin/todd/server/objects"
-	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrInvalidDBPlugin = errors.New("Invalid DB plugin in config file")
+	ErrNotExist        = errors.New("Value does not exist")
 )
 
 // DatabasePackage represents all of the behavior that a ToDD database plugin must support
 type DatabasePackage interface {
 
 	// (no args)
-	Init()
+	Init() error
 
 	// (agent advertisement to set)
-	SetAgent(defs.AgentAdvert)
+	SetAgent(defs.AgentAdvert) error
 
-	GetAgent(string) defs.AgentAdvert
-	GetAgents() []defs.AgentAdvert
+	GetAgent(string) (*defs.AgentAdvert, error)
```

Comment:
I frequently return pointers to avoid copying a struct, though there are pros and cons to that.

Since the reason is to not copy memory, and slices being a pointer to an underlying array, something like `*[]defs.AgentAdvert` doesn't make much sense. Your specific question about `[]*defs.AgentAdvert`, while there are some advantages to a slice of pointers, it doesn't make a difference as far as the return is concerned.